cpp/src/strings/json/json_path.cu

```diff
@@ -314,9 +328,25 @@ class json_state : private parser {
     if (result == parse_result::SUCCESS) { parent_el_type = prev_el_type; }
     return result;
   }
-
+  // required since we are using string_view with host data rather than device data
+  EXECUTION_SPECIFIER int string_view_compare(string_view const str1, string_view const str2)
+  {
+#if defined(HOST_DEBUGGING)
+    int idx = 0;
+    for (; (idx < str1.size_bytes()) && (idx < str2.size_bytes()); idx++) {
+      if (*(str1.data() + idx) != *(str2.data() + idx)) {
+        return *(str1.data() + idx) - *(str2.data() + idx);
+      }
+    }
+    if (idx < str1.size_bytes()) return 1;
+    if (idx < str2.size_bytes()) return -1;
+    return 0;
+#else
+    return str1.compare(str2);
+#endif
```

Comment:
I assume the `HOST_DEBUGGING` paths will be removed once the code has been debugged and before this PR goes to review?
I don't want us to have to maintain this code path in the repo. Also, I feel like it is some violation of the one-definition rule (ODR).

Reply:
I had Srikar add this specifically so that it would be included.  Essentially, I had added a similar thing to the original implementation that got removed.  But it was needed again when debugging/adding new features, so we had to reproduce it.  I think it makes sense to keep this in specifically so that we can avoid that in the future.

Reply:
As far as the #ifdef, wanted to avoid modifying string_view itself because by definition it is a wrapper around device data.  So having the compare function marked as __host__ would be somewhat weird.

Reply:
What if we refactored the guts of `string_view::compare() ` such that it became a global `__host__ __device__` function called by both string_view and this code? That way we keep string view device specific and we eliminate the duplication.

Reply:
Actually, the `string_view::compare()` function is something I've specifically done some investigation on improving using CUDA-specific intrinsic functions. I would not be in favor of making this a host/device function.

Reply:
Then I really think we should leave this as-is.  This is a piece of practical engineering.  No one who uses cudf will ever be affected by this code breaking unless they've specifically flipped this #define on to manually debug things.  And if we remove it entirely then anyone who wants to debug has to reproduce all of it anyway.

Reply:
I disagree. There are several ways to debug as you know. You picked a hacky technique of re-implementing everything in CPU code. If I had to debug this, I would likely use some other technique and this code would frankly be in my way. Nothing wrong with hacky debug code but I feel this should be removed once the code is working.